apps/aecore/src/aec_block_candidate.erl

```diff
@@ -183,14 +184,16 @@ int_create_block(PrevBlockHash, PrevBlock, Trees, Miner, Txs) ->
     Height = PrevBlockHeight + 1,
     Version = aec_hard_forks:protocol_effective_at_height(Height),
 
-    {ok, Txs1, Trees2, Fees} =
+    {ok, ValidTxs, InvalidTxs, Trees2, Fees} =
         int_apply_block_txs(Txs, Miner, Trees, Height, Version, false),
 
-    TxsTree = aec_txs_trees:from_txs(Txs1),
+    aec_tx_pool:invalid_txs([ aetx_sign:hash(STx) || STx <- InvalidTxs ], Height),
```

Comment:
Any reasons why `aec_tx_pool:invalid_txs` is not called also in code path around block candidate update?

Reply:
Small (not-obvious) optimization - garbage collect is only run after `create` thus there is no point in adding invalid_txs at this point. If they are still invalid in `create` they will be added, and otherwise we did save some work!